x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/described_form_field.tsx

```diff
@@ -3,24 +3,91 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { get } from 'lodash';
+import React, { FunctionComponent, useState } from 'react';
+import {
+  EuiDescribedFormGroup,
+  EuiDescribedFormGroupProps,
+  EuiSpacer,
+  EuiSwitch,
+  EuiSwitchProps,
+} from '@elastic/eui';
 
-import React, { FunctionComponent } from 'react';
-import { EuiDescribedFormGroup, EuiDescribedFormGroupProps } from '@elastic/eui';
+import { UseField, ToggleField, useFormData } from '../../../../shared_imports';
 
-import { ToggleableField, Props as ToggleableFieldProps } from './toggleable_field';
+interface SwitchProps extends Omit<EuiSwitchProps, 'checked' | 'onChange'> {
+  initialValue?: boolean;
+  path?: string;
+  keepChildrenMounted?: boolean;
+  onChange?: (nextValue: boolean) => void;
+}
 
 type Props = EuiDescribedFormGroupProps & {
-  switchProps: ToggleableFieldProps;
+  switchProps: SwitchProps;
+  keepChildrenMounted?: boolean;
 };
 
 export const DescribedFormField: FunctionComponent<Props> = ({
   children,
   switchProps,
+  keepChildrenMounted = false,
   ...restDescribedFormProps
 }) => {
+  const { initialValue, value, path, ...restSwitchProps } = switchProps;
+  const [uncontrolledValue, setUncontrolledValue] = useState<boolean>(
+    () => switchProps.initialValue ?? false
+  );
+  const [formData] = useFormData();
+  const controlledValue = path ? get(formData, path) : undefined;
+
+  const isContentVisible = Boolean(controlledValue ?? uncontrolledValue);
+
+  const renderChildren = () => {
+    if (keepChildrenMounted) {
+      return <div style={{ display: isContentVisible ? 'block' : 'none' }}>{children}</div>;
+    }
+    return isContentVisible ? children : null;
+  };
+
+  const renderToggle = () =>
+    path ? (
+      <UseField
+        path={path}
+        component={ToggleField}
+        componentProps={{
+          euiFieldProps: {
+            ...restSwitchProps,
+          },
+        }}
+      />
+    ) : (
+      <EuiSwitch
+        {...restSwitchProps}
+        checked={isContentVisible}
+        onChange={(e) => {
+          const nextValue = e.target.checked;
+          if (typeof controlledValue !== 'boolean') {
+            setUncontrolledValue(nextValue);
```

Comment:
`controlledValue` should be `undefined` since we don't have `path` (line 53 and line 41), so my guess is that we don't need the `typeof` check.